src/main/java/org/testng/internal/TestMethodWorker.java

```diff
@@ -53,13 +55,22 @@ public TestMethodWorker(IInvoker invoker,
                           List<IClassListener> listeners)
   {
     m_invoker = invoker;
-    m_methodInstances = testMethods;
     m_suite = suite;
     m_parameters = parameters;
     m_groupMethods = groupMethods;
     m_classMethodMap = classMethodMap;
     m_testContext = testContext;
     m_listeners = listeners;
+    List<IMethodInstance> localMethodInstances = Lists.newArrayList(testMethods);
```

Comment:
@juherr - I think I may have fixed the ordering properly as part of [this](https://github.com/cbeust/testng/pull/1345/commits/f67cd34170d0daacce7df03a3a7c10a727563656) commit in PR #1345 

Can we first get #1345 merged, and after that I will test out this without sorting and see if that solves the problem ?